lighthouse-service/event_handler/handler.go

```diff
@@ -33,6 +35,8 @@ func NewEventHandler(ctx context.Context, event cloudevents.Event) (EvaluationEv
 		return nil, err
 	}
 
+	eventStore := processEventStore(es, keptnHandler)
```

Comment:
what is this doing ? why?

Reply:
due to testability, if the es is nil (we don't pass the mocked EventStore), we should use the one from keptnHandler, if it's not nil (we passed the mocked implementation), we should use the mocked implementation

Reply:
If it's just about testing, this should not be part of the business logic (it is like introducing a `if mock {}` logic). Would be better to just pass the correct implementation of EventStore as a dependency. Also, if i delete this line, all the tests are still green

Reply:
I could not find a solution how to inject this nicely, as the EventStore is a part of keptnHandler which is created in the NewEventHandler(), which is called in OnEvent() and depends on context. If I  try to inject it from OnEvent(), I cannot get the appropiate data from context, so I get an error of `could not get eventSender from context`. Can you maybe propose some solution?

Reply:
Instead of passing in an `EventStore` value which the constructor logic needs to evaluate whether it is nil or not, you can pass in a EventStore _Provider_ that returns the correct event store. The provider is a simple function that looks like this:
```
type EventStoreProvider func(k Keptn) EventStore
``` 
now, in the test you pass a provider that just ignores the `Keptn` parameter and returns the mocked event store. In the real implementation you pass a provider provider that returns `k.EventHandler`

Reply:
resolved